lib/lintPostcssResult.js

```diff
@@ -131,4 +135,11 @@ function isFixCompatible({ stylelint }) {
 	return true;
 }
 
-module.exports = lintPostcssResult;
+/**
+ * @param {PostcssResult} result
+ * @param {string} ruleName
+ * @returns {void}
+ */
+function warnDeprecatedRule(result, ruleName) {
+	result.warn(`The "${ruleName}" rule is deprecated.`, { stylelintType: 'deprecation' });
```

Comment:
@ybiquitous

This PR was merged before #7550.
Should we use `process.emitWarning` instead?
i.e. `--no-deprecation` doesn't suppress these warnings

Reply:
Good point. We must consider backward compatibility if switching to `process.emitWarning` would make sense.

For example, we may need to update the following document:
https://github.com/stylelint/stylelint/blob/1ba1a0aeafaa124a4987a0135d511111cec3af0a/docs/developer-guide/rules.md?plain=1#L296-L301

Can you open a new issue for the discussion?

Reply:
I considered this not important since the deprecation was added 2 majors ago.
But [this](https://deps.dev/npm/stylelint/16.7.0/versions) made me realize that a lot dependents are still stuck on old majors.
e.g. WordPress/gutenberg#48142

> Can you open a new issue for the discussion?

OK.